core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -901,6 +916,282 @@ public TableToken lockTableName(CharSequence tableName, String dirName, int tabl
         return tableNameRegistry.lockTableName(tableNameStr, dirName, tableId, isWal);
     }
 
+    public void metadataCacheClear() {
+        cairoTables.clear();
+    }
+
+    /**
+     * Gets the current table metadata object from the list.
+     * <p>
+     * Returns a table if present.
+     * Returns null if table not present or could not be re-hydrated after a metadata change.
+     *
+     * @param tableToken A current table token for the table.
+     * @return The cached table metadata.
+     */
+    public CairoTable metadataCacheGetTable(@NotNull TableToken tableToken) {
+        final CairoTable table = cairoTables.get(tableToken.getDirName());
+        if (table == null) {
+            metadataCacheHydrateTable(tableToken, false, true);
+        }
+        return cairoTables.get(tableToken.getDirName());
+    }
+
+    /**
+     * This is dangerous and may clobber any concurrent metadata changes. This should only be used in last-resort cases,
+     * for example, for testing purposes.
+     */
+    public void metadataCacheHydrateAllTables() {
+        ObjHashSet<TableToken> tableTokensSet = tlTokens.get();
+        getTableTokens(tableTokensSet, false);
+        ObjList<TableToken> tableTokens = tableTokensSet.getList();
+        TableToken tableToken = tableTokens.getQuick(0);
+
+        try {
+            for (int i = 0, n = tableTokens.size(); i < n; i++) {
+                tableToken = tableTokens.getQuick(i);
+                if (!tableToken.isSystem()) {
+                    metadataCacheHydrateTable(tableTokens.getQuick(i), true, true);
+                }
+            }
+        } catch (CairoException ex) {
+            LOG.error().$("could not hydrate metadata: [table=").$(tableToken).I$();
+        } finally {
+            tlTokens.get().clear();
+        }
+    }
+
+    /**
+     * Hydrates metadata cache for a table, using thread-local Path/ColumnVersionReader objects.
+     *
+     * @param token
+     * @param blindUpsert
+     * @see CairoEngine#metadataCacheHydrateTable(TableToken, Path, ColumnVersionReader, boolean, boolean)
+     */
+    public void metadataCacheHydrateTable(@NotNull TableToken token, boolean blindUpsert, boolean infoLog) {
+        LOG.debug().$("hydrating table using thread-local path and column version reader [table=").$(token.getTableName()).I$();
+        try {
+            metadataCacheHydrateTable(token, tlPath.get(), tlColumnVersionReader.get(), blindUpsert, infoLog);
+        } finally {
+            tlPath.get().close();
+            tlColumnVersionReader.get().close();
+        }
+    }
+
+    /**
+     * Hydrates table metadata, bypassing TableWriter/Reader.
```

Comment:
this method could be private, its JavaDoc can be moved to the public method that is actually being called